spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/core/convert/TestDatastoreItemCollections.java

```diff
@@ -0,0 +1,83 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.datastore.core.convert;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * @author Dmitry Solomakha
+ */
+public class TestDatastoreItemCollections {
+	private List<Integer> intList;
+
+	private ImmutableSet<Double> doubleSet;
+
+	private String[] stringArray;
+
+	private boolean[] boolArray;
+
```

Comment:
Could you add a property like `byte[][]` or `List<byte[]>` just to make sure the converter (to Blob) is used for the component type and also that we don't try to treat `byte[]` like an array?